components/string.rst

```diff
@@ -0,0 +1,378 @@
+
+    // use the unwrap method to make the inverse conversion
+    $contents = UnicodeString::unwrap([
+        new UnicodeString('hello'), new UnicodeString('world'),
+    ]); // $contents = ['hello', 'world']
+
+There are two shortcut functions called ``b()`` and ``u()`` to create
+``ByteString`` and ``UnicodeString`` objects::
+
+    // ...
+    use function Symfony\Component\String\b;
+    use function Symfony\Component\String\u;
+
+    // both are equivalent
+    $foo = new ByteString('hello');
+    $foo = b('hello');
+
+    // both are equivalent
+    $baz = new UnicodeString('hello');
+    $baz = u('hello');
+
+There are also some specialized constructors::
+
+    // ByteString can create a random string of the given length
+    $foo = ByteString::fromRandom(12);
+
+    // CodePointString and UnicodeString can create a string from code points
+    $foo = UnicodeString::fromCodePoints(2344, 2350, 2360, 2381, 2340, 2375);
+    // $foo = 'नमस्ते'
+
+Methods to Transform String Objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each string object can be transformed into the other two types of objects::
+
+    $foo = ByteString::fromRandom(12)->toCodePointString();
+    $foo = (new CodePointString('hello'))->toUnicodeString();
+    $foo = UnicodeString::fromCodePoints(104, 101, 108, 108, 111)->toByteString();
+
+Methods Related to Length and White Spaces
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // returns the number of graphemes, code points or bytes of the given string
+    (new ByteString('नमस्ते'))->length();      // 18 (bytes)
+    (new CodePointString('नमस्ते'))->length(); // 6 (code points)
+    (new UnicodeString('नमस्ते'))->length();   // 4 (graphemes)
+
+    // returns ... ?????
+    (new ByteString('नमस्ते'))->length();      // 18
+    (new CodePointString('नमस्ते'))->length(); // 4
+    (new UnicodeString('नमस्ते'))->length();   // 4
+
+    // only returns TRUE if the string is exactly an empty string (not even white spaces)
+    u('hello world')->isEmpty();  // false
+    u('     ')->isEmpty();        // false
+    u('')->isEmpty();             // true
+
+    // removes all white spaces from the start and end of the string and replaces two
+    // or more consecutive white spaces inside contents by a single white space
+    u("  \n\n   hello        world \n    \n")->collapseWhitespace(); // 'hello world'
+
+Methods to Change Case
+~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // changes all graphemes/code points to lower case (both methods are equivalent)
+    u('FOO Bar')->lower();  // 'foo bar'
+    u('FOO Bar')->folded(); // 'foo bar'
+
+    // changes all graphemes/code points to upper case
+    u('foo BAR')->upper(); // 'FOO BAR'
+
+    // changes all graphemes/code points to "title case"
+    u('foo bar')->title();     // 'Foo bar'
+    u('foo bar')->title(true); // 'Foo Bar'
+
+    // changes all graphemes/code points to camelCase
+    u('Foo: Bar-baz.')->camel(); // 'fooBarBaz'
```

Comment:
another example: `u('Foo: Bar-baz.')->camel()->title(); // 'FooBarBaz'`
this reminds me that methods can be chained, do we tell about it?